fuse_optimizers/src/optimizer.cpp

```diff
@@ -100,36 +100,64 @@ void Optimizer::loadMotionModels()
     return;
   }
   // Validate the parameter server values
-  XmlRpc::XmlRpcValue motion_model_list;
```

Comment:
The diff here is a pain, but I didn't change much. The diff ignoring white space is a little bit better, but still not great because if shows the diff between the previous code for the `TypeArray` vs the new code for the `TypeStruct`.

In brief, I did this for each loading method:
1. Rename the variable where we get the pam to remove the `_list` suffix, because now it could either be a list (`TypeArray`) or a map/dict (`TypeStruct`).
2. Remove the `if` that was checking for `!= TypeArray` and moved the exception thrown into the new `else` block.
3. Created an `if` block for `== TypeArray` with the original code, that only has the variable w/o the `_list` suffix renamed.
4. Created an `else if` block for `== TypeStruct` with the equivalent code for the struct, which is similar to the `TypeArray` one, but with subtle differences.